File: Deepseek-Coder-33b-Instruct/stage2/enter_data_directive.f90
#ifndef T1
!T1:enter data directive,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, PARAMETER :: N = 100
        INTEGER :: i
        INTEGER, DIMENSION(:), ALLOCATABLE :: host_array, device_array

        ALLOCATE(host_array(N), device_array(N))

        ! Initialize host array
        DO i = 1, N
          host_array(i) = i
        END DO

        ! Copy host array to device
        !$acc enter data copyin(host_array) create(device_array)

        ! Update device array using enter data directive
        !$acc enter data copyin(host_array(1:N))
        !$acc parallel loop present(device_array)
        DO i = 1, N
          device_array(i) = host_array(i)
        END DO
        !$acc end parallel loop
        !$acc exit data copyout(device_array)

        ! Check if device array is updated correctly
        DO i = 1, N
          IF (device_array(i) .ne. host_array(i)) THEN
            errors = errors + 1
          END IF
        END DO

        DEALLOCATE(host_array, device_array)

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM

File: Deepseek-Coder-33b-Instruct/stage2/enter_data_directive.cpp
#include <stdlib.h>
#include <stdio.h>
#include <openacc.h>

#ifndef T1
//T1:enter data directive,V:2.7-3.3
int test1(){
    int err = 0;
    const int N = 1000;
    int *a = (int*)malloc(N * sizeof(int));
    int *b = (int*)malloc(N * sizeof(int));
    int *c = (int*)malloc(N * sizeof(int));

    for (int i = 0; i < N; ++i) {
        a[i] = i;
        b[i] = i;
        c[i] = 0;
    }

    #pragma acc data copyin(a[0:N], b[0:N]) copyout(c[0:N])
    {
        #pragma acc enter data copyin(a[0:N], b[0:N]) create(c[0:N])
        #pragma acc parallel loop
        for (int i = 0; i < N; ++i) {
            c[i] = a[i] + b[i];
        }
        #pragma acc exit data copyout(c[0:N]) delete(a[0:N], b[0:N])
    }

    for (int i = 0; i < N; ++i) {
        if (c[i] != 2 * i) {
            err = 1;
            break;
        }
    }

    free(a);
    free(b);
    free(c);

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    failed = failed + test1();
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}